cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -193,7 +193,7 @@ protected void shutDown() throws Exception {
                                          com.google.common.base.Predicate<ApplicationRecord> predicate)
     throws Exception {
     List<ApplicationRecord> appRecords = new ArrayList<>();
-    for (ApplicationSpecification appSpec : store.getAllApplications(namespace)) {
+    for (ApplicationSpecification appSpec : store.getAllApplications(namespace.toEntityId())) {
```

Comment:
can ``namespace`` be ``NamespaceId`` without too many changes? If so, let;s make changes to ``ApplicationLifecycleService`` too, in this class?